### PR TITLE
Clinical Metadata Table Integration

### PR DESCRIPTION
**Pull request by dashaylan**
This PR adds a view to display clinical metadata in a tabular format, featuring sorting, aggregation and column hiding. Accompanying the table is a custom build drop down which still has a class based implementation. That will be reworked to use functional components in a later PR.

This table is not fully styled, as seen by the search bar fully stretching across the page and other miscellaneous visual oddities. However, the core the functionality of the table is sufficient.

### CODE REVIEW

File: src/views/MetadataTable.js [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+      </Row>
+    </>
+  )
+}
+
+
+
+function CreateColumns(columnNames, cb) {
+  var columnList = [];
+  console.log(columnNames);
```
Comment (haoyuanli) [inline src/views/MetadataTable.js:286]: unneeded console statements?

File: src/views/MetadataTable.js [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+  return fetch(BASE_URL + "/"+ metadata +"/search", {
+    method: "POST",
+    body: JSON.stringify({datasetId: datasetId}),
+    headers: {
+      "Content-Type": "application/json"
+    }
+  })
+    .then(function(response) { 
+      if (!response.ok) {
+        console.log(response)
```
Comment (haoyuanli) [inline src/views/MetadataTable.js:313]: unneeded console statement?

File: src/views/MetadataTable.js [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+      Aggregated: ({ value }) => `${value} `,
+    }
+    columnList.push(column)
+  }
+  cb(columnList)
+}
+
+function getMetadataData(datasetId, metadata, cb) {
+  var datasets = [];
+  console.log("Calling")
```
Comment (haoyuanli) [inline src/views/MetadataTable.js:303]: unneeded console statement?

Comment (dashaylan) [reply src/views/MetadataTable.js:303]: I knew I forgot some

**Comment by haoyuanli** [comment]
Overall looks good, but I am still trying to think how we can further break down the `src/views/MetadataTable.js`. It does seem a little difficult though, since a big part of it is the handling of a number of utilities of the table itself, as well as the html of the table, which are pretty tightly coupled together

**Comment by dashaylan** [comment]
To make it slightly more readable, I think it's easy enough to pull the main Table function into it's own file and have the rest of the supporting functions in another. But yes, that Table function itself is quite difficult to further break down.

**Comment by haoyuanli** [comment]
> To make it slightly more readable, I think it's easy enough to pull the main Table function into it's own file and have the rest of the supporting functions in another. But yes, that Table function itself is quite difficult to further break down.
> 
> Now that I have a basic backbone for what the JSX looks like, I could write up functions that create the separate chunks. It'll look a bit cleaner from the Table side of things.

Sure, that sounds good.

**Pull request merged**